content/redirect.js

```diff
@@ -37,13 +37,37 @@ const resolve = (url) => {
         continue;
       }
 
-      const redirectPairs = fs
-        .readFileSync(redirectsFilePath, "utf-8")
+      const content = fs.readFileSync(redirectsFilePath, "utf-8");
+
+      const redirectPairs = content
         .split("\n")
         .slice(1, -1)
         .map((line) => line.trim().split(/\s+/));
+
+      const normalizedPairs = new Map();
+      for (const [from, to] of redirectPairs) {
+        normalizedPairs.set(from.toLowerCase(), to.toLowerCase());
+      }
+
       for (const [from, to] of redirectPairs) {
-        redirects.set(from.toLowerCase(), to);
+        let toNormalized = to.toLowerCase();
+        if (normalizedPairs.has(toNormalized)) {
+          // It loops back on itself!
+          // console.log("ROGUE", [from, to]);
+          const chain = [toNormalized];
+          let skip = false;
+          while (toNormalized && normalizedPairs.has(toNormalized)) {
+            toNormalized = normalizedPairs.get(toNormalized);
+            chain.push(toNormalized);
+            if (chain.length > 2) {
+              // console.log("REALLY ROGUE", [from, to], chain);
+              skip = true;
+              break;
+            }
+          }
+        } else {
+          redirects.set(from.toLowerCase(), to);
+        }
```

Comment:
Do we want to collapse this to something like the following?
```suggestion
        if (!normalizedPairs.has(toNormalized)) {
          // The "to" URL is really a final destination, and doesn't
          // show up again in the redirects as a "from" URL.
          redirects.set(from.toLowerCase(), to);
        }
```

Reply:
Hmm.. I don't think we need to keep any of it. Either way, it's out of context for the problem at hand. I'll take a look tomorrow if any of it is worth keeping.